Added a Copy Icon and Copy to Clipboard Feature to Shortened URLs

KingSand08 opened the pull request:
<img width="1512" height="821" alt="Screenshot 2025-07-10 at 1 56 47 PM" src="https://github.com/user-attachments/assets/7ba03196-fe9b-4428-8c3e-69b6e3e63414" />
<img width="1512" height="823" alt="Screenshot 2025-07-10 at 1 56 59 PM" src="https://github.com/user-attachments/assets/995b1e69-a866-420d-8765-aa916247229b" />
<img width="1512" height="824" alt="Screenshot 2025-07-10 at 1 57 08 PM" src="https://github.com/user-attachments/assets/220124ef-ee0e-4016-945c-c5776991ad2a" />
<img width="1512" height="823" alt="Screenshot 2025-07-10 at 1 57 17 PM" src="https://github.com/user-attachments/assets/8c2c82b8-eeb5-4546-bb97-5fdaec0c69e3" />

KingSand08 committed:
URL Shortener now copies to user's clipboard
- Added copy icon to Overview/SVG.js
- Icon shows when url.alias is clicked
- Icon is green at frist but then changes to default color
- url.alias now copies to clipboard (no longer is a tag, but a button)

KingSand08 committed:
Added requested changes
- Added white icons permenantly to each URL
- Made green transition just the one clicked (before reverting back to the white icon)

KingSand08 committed:
Changed the icon being used.

KingSand08 committed:
Moved copy url login to function

KingSand08 committed:
Adjusted URL page for mobile view
+ Made the Created At and Times Used columns the same as their respective columns
    - Added hidden
    - Added md:table-cell

KingSand08 committed:
Made trashcan white in dark mode, kept color for light mode

KingSand08 committed:
Fixed color change bug + simplified color function
+ Removed time state variable, now relies on just the icon id being changed
+ Switched function getCopyIcon to instead be getCopyIconClassName, which just will return the className string rather than the icon itself
+ Removed copyingId and url.id check, passing to the getCopyIconClassName function instead

evanugarte commented on src/Pages/URLShortener/URLShortener.js:
```diff
@@ -143,6 +146,14 @@ export default function URLShortenerPage() {
+    const isCopying = copyingId === urlId;
+    if (isCopying) {
+      return 'transition-colors duration-500 dark:fill-green-500 fill-[#05ab00]';
+    } else {
+      return 'transition-colors duration-500 dark:fill-[#dcdcdc] fill-[#434343]';
+    }
```
```suggestion
    }
    return 'transition-colors duration-500 dark:fill-[#dcdcdc] fill-[#434343]';
```

evanugarte commented on src/Pages/URLShortener/URLShortener.js:
```diff
@@ -453,9 +473,16 @@ export default function URLShortenerPage() {
                     return (
                       <tr className='break-all !rounded md:break-keep hover:bg-white/10' key={index}>
                         <td className=''>
-                          <a className='link link-hover link-info' target="_blank" rel="noopener noreferrer" href={`${url.link}`}>
-                            {url.alias}
-                          </a>
+                          <div className='pb-2 flex flex-row gap-2'>
+                            <button className='w-fit link link-hover link-info'
+                              onClick={() => {
+                                navigator.clipboard.writeText(url.link);
+                                setCopyingId(url.id);
+                              }}>
+                              {url.alias}
+                            </button>
+                            { copyIcon(getCopyIconClassName(copyingId, url.id)) }
+                          </div>
```
do this
```js
{allUrls.map((url, index) => {
  const copyIconClassName = getCopyIconClassName(copyingId, url.id);

  // later in the react code...
  { copyIcon(copyIconClassName) }
```

KingSand08 committed:
Addressed requested changes
- Moved the getCopyIconClassName function outside of button click function, to under the url mapping instead.
- Removed else in getCopyIconClassName, instead just returning.

adarshm11 reviewed (approved):
lgtm

altran03 reviewed (approved):
W

evanugarte committed:
nothing really

evanugarte committed:
nothing really

evanugarte committed:
nothing really

evanugarte committed:
click icon to copy

evanugarte committed:
show check after copy
